ephios/core/views/accounts.py

```diff
@@ -32,6 +34,20 @@ class UserProfileListView(CustomPermissionRequiredMixin, ListView):
     permission_required = "core.view_userprofile"
     ordering = "last_name"
 
+    def get_queryset(self):
+        global_preferences = global_preferences_registry.manager()
+        categories = global_preferences["general__relevant_qualification_categories"]
+        qs = UserProfile.objects.all()
+        for category in categories:
+            qs = qs.prefetch_related(
+                Prefetch(
+                    "qualification_grants",
+                    queryset=QualificationGrant.objects.filter(qualification__category=category),
+                    to_attr=str(category.pk),
+                )
+            ).prefetch_related(f"{category.pk}__qualification")
```

Comment:
That last one should be a select related going from n:1 (grant:qualification)

Reply:
i tried that, select_related seems not be able to use prefetched stuff, it throws a field error with `Invalid field name(s) given in select_related: '2', '1'. Choices are: (none)`

Reply:
Did you try putting it on the queryset given to the Prefetch-object? I think that's what the documentation shows.

Reply:
i used 
```
qs = qs.prefetch_related(
                Prefetch(
                    "qualification_grants",
                    queryset=QualificationGrant.objects.filter(qualification__category=category),
                    to_attr=f"qualifications_for_category_{category.pk}",
                )
            ).select_related(f"qualifications_for_category_{category.pk}__qualification")
```
I dont't think that there is another possible application order as the field is "created" by the first prefetch

Reply:
Here's what I meant:

```python
qs = qs.prefetch_related(
                Prefetch(
                    "qualification_grants",
                    queryset=QualificationGrant.objects.filter(
                        qualification__category=category,
                    ).select_related(f"qualifications_for_category_{category.pk}__qualification"),
                    to_attr=f"qualifications_for_category_{category.pk}",
                )
            )
```

Reply:
crazy shit :D

Reply:
:D